src/renderer/components/dock/dock.store.ts

```diff
@@ -35,16 +35,24 @@ export enum TabKind {
   POD_LOGS = "pod-logs",
 }
 
-export interface IDockTab {
-  id?: TabId;
+export interface DockTabDescriptor {
+  id: TabId;
   kind: TabKind;
-  title?: string;
-  pinned?: boolean; // not closable
+  title: string;
+  pinned: boolean; // not closable
+
+  /**
+   * DockTabData may contain extra data
+   */
+  [key: string]: any;
 }
 
+export type CreateDockTabDescriptor = Partial<Omit<DockTabDescriptor, "kind">> & Pick<DockTabDescriptor, "kind">;
```

Comment:
Do we really need this new type?

From the quick view the typings a bit confusing too. We omit `kind` field and then picking same field again?